src/acp/rule.test.ts

```diff
@@ -0,0 +1,704 @@
+/**
+ * Copyright 2020 Inrupt Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to use,
+ * copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
+ * Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+import { jest, describe, it, expect } from "@jest/globals";
+import { asIri, createThing } from "../thing/thing";
+import {
+  addForbiddenRuleToPolicy,
+  addOptionalRuleToPolicy,
+  addRequiredRuleToPolicy,
+  getForbiddenRuleOnPolicyAll,
+  getOptionalRuleOnPolicyAll,
+  getRequiredRuleOnPolicyAll,
+  Rule,
+  setForbiddenRuleOnPolicy,
+  setOptionalRuleOnPolicy,
+  setRequiredRuleOnPolicy,
+} from "./rule";
+import { DataFactory } from "n3";
+import { Thing, ThingPersisted } from "../interfaces";
+import { Policy } from "./policy";
+
+const ACP_ANY = "http://www.w3.org/ns/solid/acp#anyOf";
+const ACP_ALL = "http://www.w3.org/ns/solid/acp#allOf";
+const ACP_NONE = "http://www.w3.org/ns/solid/acp#noneOf";
+
+const mockRule = (url: string) =>
```

Comment:
I'm confused about why this would be considered a 'mock rule'. It just seems to be creating an empty thing, so would it not make the code clearer to just use `createThing(url)` in the rest of this code rather than attempting to abstract it into a 'Rule' (unless the idea is to later add to this function to actually do something Rule-specific, such as adding a triple to the Thing it's creating stating it's a Thing of type `acp:Rule` or something...)? Even in that case, I'd still just rename this function to `createRule()`.

Reply:
The intent was indeed to add rule-specific behaviour in this function, which is done in https://github.com/inrupt/solid-client-js/pull/525.

However, for this comment as for the previous one, it is true that these functions might be useful as part of our public API. I'll create a ticket to this intent, so that it does'nt clog this (already quite large) PR.